src/groupByKey.ts

```diff
@@ -2,6 +2,13 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedByKey:GroupsMap<T> = {};
+
+  items.map((item: T) => {
+    return (groupedByKey[`${item[key]}`] = groupedByKey[`${item[key]}`]
+    || []).push(item);
+  });
```

Comment:
O no 🙈
Should use `map` when need to create a new array with new values from the current array. But you don't catch the new array and you update object that exists outside the map callback 🥲